server/api.go

```diff
@@ -931,3 +935,139 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	ctx := context.Background()
+	githubClient := p.githubConnect(*info.Token)
+
+	repositories, _, apiErr := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{})
+	if apiErr != nil {
+		mlog.Error(apiErr.Error())
+		return
+	}
+
+	resp, _ := json.Marshal(repositories)
+	w.Write(resp)
+}
+
+func (p *Plugin) createIssue(w http.ResponseWriter, r *http.Request) {
+
+	type IssueRequest struct {
+		Title       string `json:"title"`
+		Body        string `json:"body"`
+		Repo        string `json:"repo"`
+		PostId      string `json:"post_id"`
+		CurrentTeam string `json:"current_team"`
+	}
+
+	if r.Method != http.MethodPost {
+		http.Error(w, fmt.Sprintf("Request: %s is not allowed, must be POST", r.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
+		mlog.Error("Error decoding JSON body", mlog.Err(err))
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Title == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Repo == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repo name.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.CurrentTeam == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid team", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	ctx := context.Background()
+	ghIssue := &github.IssueRequest{
+		Title: &issue.Title,
+		Body:  &issue.Body,
+	}
+
+	siteUrl := p.API.GetConfig().ServiceSettings.SiteURL
+	permalink := getPermaLink(*siteUrl, issue.PostId, issue.CurrentTeam)
```

Comment:
Side note: Providing the team is not needed to create a perma link. `getPermaLink` can just be:
```go
func getPermaLink(siteUrl, postId string) string {
	return fmt.Sprintf("%v/_redirect/pl/%v", siteUrl, postId)
}
```

Reply:
Thanks, @hanzei.  I changed the function and occurrences of the function call.